src/app/inventory/BadgeInfo.tsx

```diff
@@ -103,7 +109,13 @@ export default function BadgeInfo({ item, isCapped, rating, isWishListRoll }: Pr
         </div>
       )}
       <div className={styles.primaryStat}>
-        {item.dmg && <ElementIcon element={item.dmg} />}
+        {(item.dmg && <ElementIcon element={item.dmg} />) ||
+          (item.isDestiny2() &&
+            item.masterworkInfo &&
+            item.masterworkInfo.statHash &&
+            item.masterworkInfo.statHash in armorElements && (
```

Comment:
Should we just set `dmg` based on element type when constructing the item?

Reply:
i was a little worried that might screw with some other searches by making it seem weapon-like. i do think slightly longer term maybe this should be handled during item building

Reply:
it's suddenly occurring to me that existing armor probably already does this..

Reply:
cool, on it